presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java

```diff
@@ -0,0 +1,218 @@
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", short.class.getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendByte(byte value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", byte.class.getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendDouble(double value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", double.class.getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendFloat(float value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", float.class.getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendBoolean(boolean value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", boolean.class.getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendString(String value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendByteBuffer(ByteBuffer value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendSqlDate(SqlDate value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendSqlTime(SqlTime value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendSqlTimeWithTimeZone(SqlTimeWithTimeZone value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendSqlTimestamp(SqlTimestamp value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendSqlTimestampWithTimeZone(SqlTimestampWithTimeZone value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    @Override
+    public void reset()
+    {
+        currentColumnIndex = 0;
```

Comment:
should we `checkState` that `currentColumnIndex == columnHandles.size()`?

Reply:
yes, that would be a good idea

Reply:
I actually think this belongs in `toByteArray`, not in `reset`. So that if not all columns are appended previous row values can't be duplicated.